cpp/src/rolling/grouped_rolling.cu

```diff
@@ -388,6 +394,24 @@ std::unique_ptr<column> expand_to_column(Calculator const& calc,
   return window_column;
 }
 
+/**
+ * @brief Comparator for numeric order-by columns, handling floating point NaN values.
+ */
+struct nan_aware_less {
+  template <typename T, CUDF_ENABLE_IF(not cudf::is_floating_point<T>())>
+  __device__ bool operator()(T const& lhs, T const& rhs) const
+  {
+    return thrust::less<T>{}(lhs, rhs);
+  }
+
+  template <typename T, CUDF_ENABLE_IF(cudf::is_floating_point<T>())>
+  __device__ bool operator()(T const& lhs, T const& rhs) const
+  {
+    if (std::isnan(lhs)) { return !std::isnan(rhs); }
+    return std::isnan(rhs) ? true : lhs < rhs;
```

Comment:
Returns true if one if the inputs (but not both) is a nan? That's surprising to me and I don't see how it allows strict ordering. Can you please add a comment? to doxygen above?

Reply:
I've added documentation to explain this. `NaN` values should compare equal. But `NaN` values are deemed larger than `numeric_limits::infinity()`.

The [problem](https://godbolt.org/z/9fMM5Kr44) with using `thrust::less` is:
1. `thrust::less(NaN, 10)` returns `false`
2. `thrust::less(10, NaN)` also returns `false`

This seriously messes up the binary search (`lower_bound()`, `upper_bound()`). This comparator fixes comparisons involving `NaN`.

Reply:
I still don't get it. If NaNs are supposed to be at the end of sorted arrays, why would `my_new_less(NaN, 10)` be true?

Reply:
Please let me know if the added comment clarifies the requirement.

Reply:
> why would `my_new_less(NaN, 10)` be true?

It isn't. `my_new_less(NaN, 10) == false`, just like `thrust::less<float>{}(NaN, 10)`. This is the way it should be, because `NaN < 10` is logically false.

Try the reverse:
1. Although `10` should be less than NaN, `10 < NaN` returns `false`. [Godbolt illustration](https://godbolt.org/z/Gnqq7Y1zM).
2. #1 above causes `thrust::less<float>{}(10, NaN)` to return `false`. This messes up binary search.
3. `my_new_less(10, NaN)` returns `true`.

Reply:
(I wonder if I'm not grokking your question.)

Reply:
I understand why (10, NaN) should and does return true (actual diff from thrust). The problem I see is the comparison in the other direction:
https://godbolt.org/z/YKaGcseWd
The operator returns true for(NaN, 10), but it should return false.

Reply:
Oh, wow. Sorry I missed your point. It's a good one.

I'll examine the other side as well.

Reply:
It appears the definition should be:
```c++
    if (std::isnan(lhs)) { return false; }
    return std::isnan(rhs) ? true : thrust::less<T>{}(lhs, rhs);
```
Does this look right, @vuule? It's a little confusing in that it's not symmetric.

I'm checking that this doesn't break a larger test.

Reply:
All good. I have added tests for this specifically. The CUDF `ROLLING_TEST` and the Spark tests seem to pass.